@xen-orchestra/immutable-backups/doc.md

```diff
@@ -0,0 +1,147 @@
+# Imutability
+
+the goal is to make a remote that XO can write, but not modify during the immutability duration set on the remote. That way, it's not possible for XO to delete or crypt any backup during this period. It protects your backup agains ransomware, at least as long as the attacker does not have a root access to the remote server.
+
+We target `governance` type of immutability, **the local root account of the remote server will be able to lift immutability**.
+
+We use the file system capabilities, they are tested on the protection process start.
+
+It is compatible with encryption at rest made by XO.
+
+The `protect` and `lift` processes must be run as root on the remote. On start, the protect process write into the remote `metadata.json` file its status (last protection applied, last protection lifted)
+
+## Configuring
+
+this package uses app-conf to store its config. The application name id `immutable-backup`
+
+## Making a file immutable
+
+when marking a file or a folder immutable, it create an alias file in `.immutable/<hash of full path>.<file|dir>.<extension>` containing the path to the folder/file and make this file also immutable.
```

Comment:
I'm ill-at-ease with the fact that the content of these files will not be encrypted even if the rest of remote is.

What bothers me is the inconsistency.

Possible solutions:
- use symlinks instead
- put these files in a specific directory which will only contain unencrypted files.

Reply:
* symlinks can't be made immutable `chattr: Operation not supported while reading flags on source`
* I think we don't really need to have xo use the hidden index folder